vyper/semantics/types/function.py

```diff
@@ -483,6 +483,9 @@ def implements(self, other: "ContractFunctionT") -> bool:
         if self.mutability > other.mutability:
             return False
 
+        if other.is_payable and not self.is_payable:
+            return False
+
```

Comment:
```suggestion
        if self.mutability != other.mutability:
            return False
```
I think we should go all the way and enforce 1 to 1 matching: https://github.com/vyperlang/vyper/issues/3774#issuecomment-1961161976

Reply:
at this point, just `return self.mutability == other.mutability`